flfm/batch.py

```diff
@@ -0,0 +1,125 @@
+import multiprocessing
+import shutil
+from pathlib import Path
+
+from dask.distributed import LocalCluster
+
+import flfm.util
+from flfm.settings import settings
+
+# TODO: Refactor this to some __init__.py. See https://github.com/ssec-jhu/flfm/issues/146.
+match settings.BACKEND:
+    case "torch":
+        import flfm.pytorch_io as flfm_io
+        import flfm.pytorch_restoration as flfm_restoration
+    case "jax":
+        import flfm.io as flfm_io
+        import flfm.restoration as flfm_restoration
+    case _:
+        raise NotImplementedError(f"Backend {settings.BACKEND} not implemented.")
+
+
+def do_reconstruction(psf, image_filename, output_filename, recon_kwargs, crop_kwargs, write=False):
+    """Do full reconstruction and save to file."""
+    # TODO: Move this func to same place that wrappers live for #146 & #135.
+
+    # Open light-field image.
+    image = flfm_io.open(image_filename)
+    # Reconstruct.
+    reconstruction = flfm_restoration.richardson_lucy(image, psf, **recon_kwargs)
+    # Crop.
+    cropped_reconstruction = flfm.util.crop_and_apply_circle_mask(reconstruction, **crop_kwargs)
+    # Save.
+    if write:
+        flfm_io.save(output_filename, cropped_reconstruction)
+    return cropped_reconstruction
+
+
+def batch_reconstruction(
+    input_dir: str | Path,
+    output_dir: str | Path,
+    psf,
+    n_workers: int | None = None,
+    n_threads: int = 2,
+    clobber=False,
+    recon_kwargs=None,
+    crop_kwargs=None,
+    carry_on=False,
+):
+    """
+    Batch parallel process multiple 3D reconstructions of input light-field images present in `input_dir`.
+
+    :param input_dir: Input directory.
+    :param output_dir: Output directory.
+    :param psf: Loaded PSF image.
+    :param n_workers: Numbers of parallel workers.
+    :param n_threads: Number of threads per worker.
+    :param clobber: Write over files in `output_dir`, otherwise raise if `output_dir` exists. Defaults to `False`.
+    :param recon_kwargs: kwargs passed to `richardson_lucy()`.
+    :param crop_kwargs: kwargs pass to `flfm.util.crop_and_apply_circle_mask()`.
+    :param carry_on: Whether to continue processing from a previous attempt. Will only process input files not in
+    `output_dir`.
+    :return: List of processed filenames.
+    """
```

Comment:
The doc string looks like rst style, but the other functions in the project are written in google style. This should either match the others or the others should match this. If we keep rst, I think the summary line goes on the same line as the triple-double quotes.